tests/system_test.py

```diff
@@ -360,51 +359,35 @@ def error(msg):
                     self.pid, self.name, msg, ' '.join(self.args),
                     self.outfile + '.cmd', f.read()))
 
-        # If the process is already dead, this might mean that the process has crashed (For example: a router
-        # crash). self.poll() will return a non None value if the process is already gone.
-        # A None value indicates that the process hasn’t terminated yet.
-        # A negative value -N indicates that the child was terminated by signal N (POSIX only).
-        # A positive value indicates a fail.
-        status = self.poll()
-
-        if status is None:
+        state = self.poll()
+        if state is None:
             # If status is None, it means that the process is still running
-            self.actual = Process.RUNNING
-        elif status > 0:
-            self.actual = Process.EXIT_FAIL
-        else:
-            self.actual = status
-
-        if status is None:
-            # The process is still running.
-            # Call process.terminate() which will try to stop the child by sending it a SIGTERM
-            # but is not guaranteed to stop it
             self.terminate()
             try:
-                # Wait for TIMEOUT seconds and then catch the
-                # TimeoutExpired and kill() the process.
-                self.wait(timeout=TIMEOUT)
+                self.wait(TIMEOUT)
             except TimeoutExpired:
-                # The terminate call() has failed to terminate the process even after we waited for TIMEOUT seconds.
-                # Send a SIGKILL to the child by calling kill()
-                self._logger.log("Wait timeout expired trying to terminate "
-                                 "process %s (name=%s). Trying to kill process now" % (self.pid, self.name))
                 self.kill()
```

Comment:
Not within scope of PR, but what about sending it SIGABORT instead of SIGKILL? That way, core is dumped and backtrace can be obtained to investigate where process got stuck.